src/App.js

```diff
@@ -1,13 +1,105 @@
 import React, { Component } from 'react'
+import PropTypes from 'prop-types'
 import { Instructions } from './components'
 import './styles/app.css'
 
+class Counter extends Component {
+  static propTypes = {
+    initialCount: PropTypes.number.isRequired
+  }
+  constructor(props) {
+    super(props)
+    this.state = { count: this.props.initialCount }
+  }
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      count: nextProps.initialCount
+    })
+  }
+  incClick = () => {
+    this.setState(state => ({
+      count: state.count + 1
+    }))
+  }
+  decClick = () => {
+    this.setState(state => ({
+      count: state.count - 1
+    }))
+  }
+  render() {
+    return (
+      <div>
+        Count state: {this.state.count}
+        <button onClick={this.decClick}>Decrease</button>
+        <button onClick={this.incClick}>Increase</button>
+      </div>
+    )
+  }
+}
+
+class InitialCountForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      count: 0,
```

Comment:
Do you really need to keep track of count in this component? It's not being used for anything. You can pretty much just keep `newCount`, and when clicked, you're calling `appCallback` with `newCount`. `count` seems redundant to me